### PR TITLE
implementing clean code architechture

### PR DESCRIPTION
**Pull request by johndy2742**
### 
This pull request focuses on refactoring the existing codebase to adhere to the principles of clean architecture. By restructuring the project into clean architecture components, we aim to enhance code maintainability, readability, and scalability.

### Changes made :

- Reorganized the codebase into distinct layers: Delivery, usecase, repository, and models.
- Integrate hot reload using air to see live updates

### CODE REVIEW

File: tmp/build-errors.log
Comment (iqbal167) [inline tmp/build-errors.log:1]: please ignore this file, you can create file `.gitignore` and add `/tmp` directory in git ignore file

Comment (johndy2742) [reply tmp/build-errors.log:1]: ok kang, is there anything else i can improve?

Comment (iqbal167) [reply tmp/build-errors.log:1]: that's enough @johndy2742

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Ignore tmp directory
+/tmp
```
Comment (iqbal167) [inline .gitignore]: Please add new line for end of file

Comment (johndy2742) [reply .gitignore]: ok

**Review by iqbal167 — APPROVED** [review]

**Pull request merged**